pandas/core/indexers.py

```diff
@@ -209,16 +209,21 @@ def validate_indices(indices: np.ndarray, n: int) -> None:
 
     Examples
     --------
-    >>> validate_indices([1, 2], 3)
-    # OK
-    >>> validate_indices([1, -2], 3)
-    ValueError
-    >>> validate_indices([1, 2, 3], 3)
-    IndexError
-    >>> validate_indices([-1, -1], 0)
-    # OK
-    >>> validate_indices([0, 1], 0)
-    IndexError
+    >>> validate_indices(np.ndarray([1, 2]), 3) # OK
+
+    >>> validate_indices(np.ndarray([1, -2]), 3)
+    Traceback (most recent call last):
+        ...
+    ValueError: negative dimensions are not allowed
+
+    >>> validate_indices(np.ndarray([1, 2, 3]), 3) # OK
+
+    >>> validate_indices(np.ndarray([-1, -1]), 0)
+    Traceback (most recent call last):
+        ...
+    ValueError: negative dimensions are not allowed
+
+    >>> validate_indices(np.ndarray([0, 1]), 0) # OK
```

Comment:
The docs in master ATM says it should raise an ```IndexError```, but it does not.
Would be glad if someone will take a look here

Reply:
Why are you using np.ndarray instead of np.array? The latter one raises

Reply:
> Why are you using np.ndarray instead of np.array? The latter one raises

I complety missed it, I have pushed a fix for it in https://github.com/pandas-dev/pandas/pull/40903/commits/857c11060a30191c4001188324fc50080dc22c07